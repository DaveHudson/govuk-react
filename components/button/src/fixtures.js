import React from 'react';
import styled from 'react-emotion';
import { boolean } from '@storybook/addon-knobs/react';
import { ButtonArrow } from '@govuk-react/icons';
import {
  BLUE,
  WHITE,
} from 'govuk-colours';

import Button from '.';

const CustomStyledButton = styled(Button)({
  backgroundColor: WHITE,
  borderStyle: 'solid',
  borderWidth: '2px',
  borderColor: BLUE,
  color: BLUE,
});

export default () => <Button>Save and continue</Button>;

export const ButtonCustomStyles = () => <CustomStyledButton>Save and continue</CustomStyledButton>;

export const ButtonStart = () => <Button start={boolean('Start', true)}>Save and continue</Button>;

export const ButtonStartIcon = () => (
  <Button icon={<ButtonArrow />} start={boolean('Start', true)}>
    Start now
  </Button>
);

export const ButtonDisabled = () => <Button disabled={boolean('Disabled', true)}>Disabled primary button</Button>;

export const ButtonDisabledStartIcon = () => (
  <Button
    disabled={boolean('Disabled', true)}
    start={boolean('Start', true)}
    icon={<ButtonArrow />}
  >
    Start now
  </Button>
);
