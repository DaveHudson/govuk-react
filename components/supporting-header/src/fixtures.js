import React from 'react';
import { text } from '@storybook/addon-knobs/react';

import SupportingHeader from '.';

const exampleHeading = 'Heading text';

const SupportingHeaderWithKnobs = () => (
  <SupportingHeader>{text('children', exampleHeading)}</SupportingHeader>
);

export default SupportingHeader;

export {
  exampleHeading,
  SupportingHeaderWithKnobs,
};
