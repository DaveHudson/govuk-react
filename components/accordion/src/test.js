import React from 'react';
import { cleanup, fireEvent, render } from 'react-testing-library';
import { YELLOW } from 'govuk-colours';
import { FOCUS_WIDTH } from '@govuk-react/constants';

import 'jest-dom/extend-expect';

import { ExampleWithoutSummaries } from './fixtures';

const focusedStyle = `
  outline: ${FOCUS_WIDTH} solid ${YELLOW};
  outline-offset: 0;
`;

afterEach(cleanup);

describe('Tabs', () => {
  it('clicking button changes focus on header', async () => {
    const { container } = render(<ExampleWithoutSummaries
      initialState={{
        accordionOne: false,
        accordionTwo: false,
        accordionThree: false,
        accordionFour: false,
      }}
    />);
    const firstButton = container.querySelector('button');
    const secondButton = container.querySelectorAll('button')[1];

    // const firstHeader = container.querySelector('div div div div:nth-child(2) div');
    const firstHeader = container.querySelector('div:nth-child(2) > div');
    const secondHeader = container.querySelector('div:nth-child(3) > div');

    expect(firstHeader).not.toHaveStyle(focusedStyle);

    // represents click event, jsdom handles focus onClick differently to browser
    fireEvent.focus(firstButton);

    expect(firstHeader).toHaveStyle(focusedStyle);

    // represents second click event, jsdom handles focus onClick differently to browser
    fireEvent.blur(firstButton);
    fireEvent.focus(secondButton);

    expect(firstHeader).not.toHaveStyle(focusedStyle);
    expect(secondHeader).toHaveStyle(focusedStyle);
  });

  it('clicking button toggles visible children', () => {
    const { container, queryByText } = render(<ExampleWithoutSummaries
      initialState={{
        accordionOne: false,
        accordionTwo: false,
        accordionThree: false,
        accordionFour: false,
      }}
    />);
    const firstButton = container.querySelector('button');
    fireEvent.click(firstButton);

    const children = queryByText('This is the content for writing well for the web.');

    expect(children).toBeInTheDocument();

    fireEvent.click(firstButton);

    expect(children).not.toBeInTheDocument();
  });

  it('matches wrapper snapshot', () => {
    const { asFragment } = render(<ExampleWithoutSummaries
      initialState={{
        accordionOne: false,
        accordionTwo: false,
        accordionThree: false,
        accordionFour: false,
      }}
    />);
    expect(asFragment()).toMatchSnapshot();
  });

  // TODO: test for clicking between tabs above media Queries width.
  // Not currently possible with jsdom
});
