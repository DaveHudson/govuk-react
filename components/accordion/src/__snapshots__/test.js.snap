// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs matches wrapper snapshot 1`] = `
<DocumentFragment>
  .c0 {
  padding-top: 30px;
  text-align: center;
}

.c1 {
  max-width: 960px;
  margin-left: 15px;
  margin-right: 15px;
  text-align: left;
}

.c8 {
  position: absolute;
  top: 50%;
  right: 15px;
  width: 16px;
  height: 16px;
  margin-top: -8px;
}

.c8:after,
.c8:before {
  content: "";
  box-sizing: border-box;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 25%;
  height: 25%;
  margin: auto;
  border: 2px solid transparent;
  background-color: #0b0c0c;
}

.c8:before {
  width: 100%;
}

.c8:after {
  height: 100%;
}

.c4 {
  border-top: 1px solid #bfc1c3;
  padding-top: 0;
}

.c5 {
  position: relative;
  padding-right: 40px;
  padding-bottom: 15px;
  cursor: pointer;
}

.c5:hover {
  background-color: #f8f8f8;
}

.c6 {
  margin-top: 0;
  margin-bottom: 0;
}

.c7 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 18px;
  line-height: 1.1111111111111112;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 100%;
  margin-top: 0;
  margin-bottom: 0;
  margin-left: 0;
  padding-top: 15px;
  padding-bottom: 0;
  padding-left: 0;
  border-width: 0;
  color: #005ea5;
  background: none;
  text-align: left;
  cursor: pointer;
}

.c7:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
  background-color: #ffbf47;
}

.c7:focus {
  outline: none;
  background: none;
}

.c7:after {
  content: "";
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c3 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 14px;
  line-height: 1.1428571428571428;
  display: inline;
  border-width: 0;
  color: #005ea5;
  background: none;
  cursor: pointer;
}

.c3:link {
  color: #005ea5;
}

.c3:visited {
  color: #4c2c92;
}

.c3:hover {
  color: #2b8cc4;
}

.c3:active {
  color: #2b8cc4;
}

.c3:focus {
  color: #0b0c0c;
}

.c3:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
  background-color: #ffbf47;
}

.c3:focus {
  background: none;
}

.c2 {
  margin-bottom: 20px;
  border-bottom: 1px solid #bfc1c3;
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-left: 30px;
    margin-right: 30px;
  }
}

@media only screen and (min-width:1020px) {
  .c1 {
    margin: 0 auto;
  }
}

@media (hover:none) {
  .c5:hover {
    background-color: initial;
  }
}

@media print {
  .c7 {
    font-size: 18px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c7 {
    font-size: 24px;
    line-height: 1.25;
  }
}

@media print {
  .c7 {
    font-family: sans-serif;
  }
}

@media print {
  .c3 {
    font-family: sans-serif;
  }
}

@media print {
  .c3[href^="/"]::after,
  .c3[href^="http://"]::after,
  .c3[href^="https://"]::after {
    content: " (" attr(href) ")";
    font-size: 90%;
    word-wrap: break-word;
  }
}

@media print {
  .c3 {
    font-size: 14px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    font-size: 16px;
    line-height: 1.25;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    margin-bottom: 30px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          style="text-align: right;"
        >
          <a
            class="c3"
            href="#"
          >
            Open all
          </a>
        </div>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <h2
              class="c6"
            >
              <button
                class="c7"
                type="button"
              >
                Writing well for the web
              </button>
            </h2>
            <div
              class="c8"
            />
          </div>
        </div>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <h2
              class="c6"
            >
              <button
                class="c7"
                type="button"
              >
                Writing well for specialists
              </button>
            </h2>
            <div
              class="c8"
            />
          </div>
        </div>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <h2
              class="c6"
            >
              <button
                class="c7"
                type="button"
              >
                Know your audience
              </button>
            </h2>
            <div
              class="c8"
            />
          </div>
        </div>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <h2
              class="c6"
            >
              <button
                class="c7"
                type="button"
              >
                How people read
              </button>
            </h2>
            <div
              class="c8"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
